### ISSUE TITLE
Maximum recursion error during clustering step

### ISSUE THREAD

**Issue by joerivstrien** [joerivstrien/process_maxquant#6]
When running the tool on a protein_groups file with 6 samples, (each having 60 slices) an error occurs during the clustering step.

The GUI displays the following error:
An exception occurred while applying clustering on a sample
Maximum recursion depth exceeded in comparison.

relevant lines in log file before error occurs:
INFO:root:Step 4, cluster the fractions per sample using hierarchical clustering.
INFO:root:Start hierarchical clustering for sample T5

(not sure why it starts clustering T5? maybe something went wrong with parsing the other samples? Samples are T1 - T6)

**Comment by joerivstrien** [comment]
Turns out T5 is also the first sample in the output (the order of samples in the output is random I guess?), so thats why it started with T5. In the end clustering worked for 2 out of 6 samples, for all the other samples maximum recursion depth error occurred.

Nice that the program continues anyway and produces an output with those that succeeded!

I can look into why the clustering code produces this error?

**Comment by ArielKomen** [comment]
Using google I found out that the implementation of hierarchical clustering of scipy is mainly used for dendogram visualizations and thus isn't able to compute large dendograms. I found [this implementation](https://calvinmccarter.wordpress.com/2014/02/20/scaling-up-hierarchical-clustering/) and applied in the code, tested it and the clustering step has become faster and can cluster more proteins than before. So, this issue has been solved.
